src/media/js/elements/select.js

```diff
@@ -0,0 +1,218 @@
+                    }
+                    root.appendChild(mktOptGroup);
+                    mktOptGroup.optGroup = optGroup;
+                    select.appendChild(optGroup);
+
+                    root.appendChild(select);
+                    root.select = select;  // Set select element.
+
+                    updateMktOptionIndices(
+                        root, root.querySelector('mkt-option[selected]'));
+
+                    // Align options with <mkt-selected-text>.
+                    if (alignMktOptions(root) > 0) {
+                        root.setAttribute('data-mkt-select--aligned', true);
+                    }
+                }
+            },
+            toggle: {
+                // Toggles visibility of the <mkt-option>s.
+                value: function(bool) {
+                    if (!this.getAttribute('data-mkt-select--aligned')) {
+                        if (alignMktOptions(this) > 0) {
+                            this.setAttribute('data-mkt-select--aligned', true);
+                        }
+                    }
+                    if (bool) {
+                        this.classList.toggle(VISIBLE, bool);
+                    } else {
+                        this.classList.toggle(VISIBLE);
+                    }
+                    return this;
+                },
+            },
+            select: {
+                // Value is set in the createdCallback.
+                value: null,
+                writable: true
+            }
+        }),
+    });
+
+    function alignMktOptions(mktSelect) {
+        var offset = $('mkt-selected-text', mktSelect).position().left;
+        $('mkt-option', mktSelect).css('padding-left', offset);
+        return offset;
+    }
+
+    function updateMktOptionIndices(mktSelect, selectedOption) {
+        // Sets a data-index on each mkt-option that is not [selected].
+        // Prominently used to set colors since we don't have CSS4 nth-match.
+        var mktOptions = mktSelect.querySelectorAll('mkt-option');
+        var index = 0;
+        for (var i = 0; i < mktOptions.length; i++) {
+            var mktOption = mktOptions[i];
+            if (mktOption === selectedOption) {
+                mktOption.removeAttribute('data-mkt-option--index');
+            } else {
+                mktOption.setAttribute('data-mkt-option--index', index++);
+            }
+        }
+    }
+
+    function copyAttrs(dest, source) {
+        // Copies attributes from one element to another.
+        for (var i = 0; i < source.attributes.length; i++) {
+            var attr = source.attributes[i];
+            if (attr !== 'id') {
+                dest.setAttribute(attr.name, attr.value);
+            }
+        }
+        return dest;
+    }
+
+    function proxyElement(htmlElement, customElement, key) {
+        // Proxy custom element interface to actual element.
+        Object.getOwnPropertyNames(htmlElement.prototype).forEach(function(method) {
```

Comment:
I feel like this should accept the objects that it is proxying not their "class". So this should just be `htmlElement` not `htmlElement.prototype` and the when it is called it should be called with `HTMLSelectElement.prototype`.

It might not matter since this probably won't be reused but it would make it more reusable.